src/fondant/core/manifest.py

```diff
@@ -196,13 +143,41 @@ def copy(self) -> "Manifest":
     def metadata(self) -> t.Dict[str, t.Any]:
         return self._specification["metadata"]
 
+    @property
+    def index(self) -> t.Dict[str, t.Any]:
+        return self._specification["index"]
+
     def update_metadata(self, key: str, value: t.Any) -> None:
         self.metadata[key] = value
 
     @property
     def base_path(self) -> str:
         return self.metadata["base_path"]
 
+    def get_dataset_location(self, spec: ComponentSpec) -> str:
```

Comment:
Can you make this a property?

Reply:
Not really. It is used later on in `data_io.py`

```python
location = self.manifest.get_dataset_location(self.component_spec)
```

We use the component_spec to retrieve the dataset location. `base_path + spec.component_folder_name`
I thought it would be nice to encapsulate this logic into a method, to give this a name. 
But maybe the manifest isn't the right place for it.